### ISSUE TITLE
Tiebreaker for case insensitive search

### ISSUE THREAD

**Issue by m-chaturvedi** [junegunn/fzf#1755]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [x] zsh
    - [ ] fish
    
## Problem / Steps to reproduce

When using case insensitive search, we don't have an option to sort depending on exact match.  Ex:

`echo "hello\nHello\nhEllo\nheLlo\nhelLo\nhellO" | fzf -i -q 'HeLlo'` 

gives the output in the same order as the input.
What about having an option which makes sure that exact match (case-sensitive), if found should be at the top?

**Comment by junegunn** [comment]
Any reason you don't use the default smart-case match? i.e. Case-insensitive by default, case-sensitive when the query contains one or more upper-case characters. It should be good enough for most use cases.

**Comment by m-chaturvedi** [comment]
> Any reason you don't use the default smart-case match? [...]

Yeah, I could, but it doesn't solve the problem when my query contains an exact match. Philosophically, in our smart-case search, we're "favoring" upper-case over lower-case, which is true mostly, because in general we type more letters small than capital, but our smart-caps might miss the exact match, like here:

`echo "Hello\nhEllo\nheLlo\nhelLo\nhellO\nhello" | fzf -q 'hello'`

(The example is, of course, pathological, but the behavior had actually happened when I was querying something.  I just wanted to point it out, because I like the tool.)
